src/components/ScrollButton.tsx

```diff
@@ -0,0 +1,34 @@
+"use client";
+
+import { useState } from "react";
+
+import styles from "./ScrollButton.module.css";
+
+export function ScrollButton() {
+	const [visible, setVisible] = useState(false);
+
+	const toggleVisible = () => {
+		const scrolled = document.body.scrollTop;
+		setVisible(scrolled > 100);
+	};
+
+	const scrollToTop = () => {
+		document.body.scrollTo({
+			behavior: "smooth",
+			top: 0,
+		});
+	};
+
+	document.body.addEventListener("scroll", toggleVisible, { passive: true });
+
+	return (
+		<button
+			className={styles.scrollButton}
+			onClick={scrollToTop}
+			style={{ display: visible ? "flex" : "none" }}
```

Comment:
[Cleanup] Hmm, I don't like mixing inline style objects with classes. Let's solidify on classes.

This will add the `styles.visible` class only if `visible` is truthy (`import clsx from 'clsx';`):

```ts
className={clsx(styles.scrollButton, visible && styles.visible)}
```

...and in the CSS file:

```css
.scrollButton {
  /* ... */
  display: none;
}

.visible {
  display: flex;
}
```